### PR TITLE
feat(cli): support non-active template version pushes

### PR DESCRIPTION
**Pull request by goodspark**
So that we can push template updates for testing without impacting normal users of the template.

### CODE REVIEW

**Comment by matifali** [comment]
Hi, @goodspark  A nice contribution. What do you think about changing the arg to `--not-active` or `--no-active` then we can keep the default behavior where pushing an update makes the pushed version active and will also allow this new use case.

A lot of people use push in ci and changing the default behaviour will break the ci.

File: cli/templatepush.go [hunk 1]
```diff
@@ -233,6 +236,12 @@ func (r *RootCmd) templatePush() *clibase.Cmd {
 			Description: "Always prompt all parameters. Does not pull parameter values from active template version.",
 			Value:       clibase.BoolOf(&alwaysPrompt),
 		},
+		{
+			Flag:        "make-active",
```
Comment (ammario) [inline cli/templatepush.go:240]: Let's make this `activate`, and the variables too.

**Review by ammario — CHANGES_REQUESTED** [review]
Just a small request. Also, please run `make update-golden-files gen`. Thanks!

File: cli/templatepush.go (hunk 1 above)
Comment (goodspark) [reply cli/templatepush.go]: When I run make, I get errors that seem unrelated to this change:

<details>
<summary>Errors
</summary>

```
# github.com/cenkalti/backoff/v4
../../go/pkg/mod/github.com/cenkalti/backoff/v4@v4.2.0/retry.go:10:26: syntax error: unexpected any, expecting ]
../../go/pkg/mod/github.com/cenkalti/backoff/v4@v4.2.0/retry.go:10:38: method has no receiver
../../go/pkg/mod/github.com/cenkalti/backoff/v4@v4.2.0/retry.go:16:53: syntax error: unexpected [ after top level declaration
../../go/pkg/mod/github.com/cenkalti/backoff/v4@v4.2.0/retry.go:17:16: method has no receiver
../../go/pkg/mod/github.com/cenkalti/backoff/v4@v4.2.0/retry.go:17:16: syntax error: unexpected (, expecting name or (
../../go/pkg/mod/github.com/cenkalti/backoff/v4@v4.2.0/retry.go:42:19: syntax error: unexpected [, expecting (
../../go/pkg/mod/github.com/cenkalti/backoff/v4@v4.2.0/retry.go:53:25: syntax error: unexpected [, expecting (
../../go/pkg/mod/github.com/cenkalti/backoff/v4@v4.2.0/retry.go:66:33: syntax error: unexpected [, expecting (
../../go/pkg/mod/github.com/cenkalti/backoff/v4@v4.2.0/retry.go:70:19: syntax error: unexpected [, expecting (
../../go/pkg/mod/github.com/cenkalti/backoff/v4@v4.2.0/retry.go:76:2: syntax error: non-declaration statement outside function body
../../go/pkg/mod/github.com/cenkalti/backoff/v4@v4.2.0/retry.go:76:2: too many errors
note: module requires Go 1.18
# github.com/coder/coder/cryptorand
cryptorand/slices.go:9:6: missing function body
cryptorand/slices.go:9:13: syntax error: unexpected [, expecting (
note: module requires Go 1.20
Makefile:473: recipe for target 'coderd/database/dump.sql' failed
make: *** [coderd/database/dump.sql] Error 2
```

</details>

**Comment by ammario** [comment]
@goodspark it looks like you're on a pre-generics version of the Go toolchain. Try rerunning the commands with go 1.20.

**Review by ammario — COMMENTED** [review]
I'm also happy to take this to the finish line if you'd like. Unfortunately GitHub makes it hard for us to push to forks.

**Comment by goodspark** [comment]
Yeah, sure. My env has some old version of Golang and since I rarely use it at work, I haven't been bothered enough to download the 1.20.

File: cli/templatepush_test.go [hunk 2]
```diff
@@ -156,6 +156,52 @@ func TestTemplatePush(t *testing.T) {
...
+		_ = coderdtest.AwaitTemplateVersionJob(t, client, version.ID)
+
+		template := coderdtest.CreateTemplate(t, client, user.OrganizationID, version.ID)
+
+		// Test the cli command.
+		source := clitest.CreateTemplateVersionSource(t, &echo.Responses{
+			Parse:          echo.ParseComplete,
+			ProvisionApply: echo.ProvisionComplete,
+		})
+		inv, root := clitest.New(t, "templates", "push", template.Name, "--make-active", "false", "--directory", source, "--test.provisioner", string(database.ProvisionerTypeEcho), "--name", "example")
```
Comment (matifali) [inline cli/templatepush_test.go:173]:
```suggestion
		inv, root := clitest.New(t, "templates", "push", template.Name, "--activate", "false", "--directory", source, "--test.provisioner", string(database.ProvisionerTypeEcho), "--name", "example")
```

**Review by ammario — APPROVED** [review]

**Pull request merged**